### PR TITLE
chore(skills): enforce required frontmatter metadata

### PR DESCRIPTION
**Pull request by anastasia-nesterenko**
## Summary

Adds local pre-commit validation that rejects new or modified canonical `SKILL.md` files when `metadata.author` is not a non-empty string or `allowed-tools` is missing or invalid. Backfills the required fields across existing canonical skills so the hook starts clean.

## Changes

- Add `.github/scripts/check_skill_frontmatter.py` with strict author/tool validation.
- Add regression tests, including numeric, boolean, collection, empty, and whitespace-only author values.
- Add the checker as a local pre-commit hook for changed `SKILL.md` files.
- Document the required skill metadata and backfill it across canonical skills.

## Type of Change

- [ ] Code change (feature, bug fix, or refactor)
- [ ] Code change with documentation updates
- [ ] Documentation only
- [x] Contributor tooling or automation
- [ ] CI, build, or test infrastructure

## Quality Gates

- [x] Tests added or updated for changed behavior
- [ ] Existing tests cover changed behavior — justification:
- [ ] Tests not applicable — justification:
- [x] Documentation updated for user-visible behavior
- [ ] Documentation not applicable — justification:

## Verification

- [x] Pull request title follows the repository's Conventional Commit format
- [x] Every commit includes an appropriate `Signed-off-by:` trailer
- [x] `uv run pre-commit run -a` passes, or any blocked checks are identified below
- [x] Targeted tests pass, or tests are marked not applicable above
- [x] No secrets, API keys, or credentials are included

Targeted validation:

- `uv run --script .github/scripts/tests/test_check_skill_frontmatter.py` — 19 passed.
- `uv run --frozen ruff check .github/scripts/check_skill_frontmatter.py .github/scripts/tests/test_check_skill_frontmatter.py` — passed.
- `uv run --frozen ruff format --check .github/scripts/check_skill_frontmatter.py .github/scripts/tests/test_check_skill_frontmatter.py` — passed.
- `uv run --script .github/scripts/check_skill_frontmatter.py --base-ref origin/main` — 44 changed skills valid.
- `flox activate --dir tools/actionlint -- actionlint` — passed.
- `uv run pre-commit run -a` — passed.

### CODE REVIEW

File: .github/scripts/check_skill_frontmatter.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+
+def check_file(path: Path) -> list[Violation]:
+    violations: list[Violation] = []
+    data = frontmatter(path)
+    if data is None:
+        return [Violation(path, "file does not open with a valid YAML frontmatter block")]
+
+    metadata = data.get("metadata")
+    author = metadata.get("author") if isinstance(metadata, dict) else None
+    if not author or not str(author).strip():
```
[the remark this answers is not in the document]
Comment (anastasia-nesterenko) [reply .github/scripts/check_skill_frontmatter.py]: Fixed in a066fa5e7a: metadata.author must now be a non-empty string, with regression coverage for numeric, boolean, list, mapping, empty, and whitespace-only values. The targeted suite passes (19 tests), and the full pre-commit suite passes.

File: plugins/nemo-eval-author/skills/eval-author-audit/SKILL.md [hunk 2]
```diff
@@ -31,7 +31,9 @@ compatibility: >-
 maturity: alpha
 license: Apache-2.0
 user-invocable: true
-allowed-tools: [Bash, Read, Write, Grep, Glob]
+allowed-tools: Bash, Read, Write, Grep, Glob
```
Comment (mckornfield) [inline plugins/nemo-eval-author/skills/eval-author-audit/SKILL.md:34]: Curious for all of these: it's not a real bash list?

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -129,6 +129,11 @@ repos:
         language: system
         stages: [commit-msg]
         always_run: true
+      - id: skill-frontmatter-lint
```
Comment (mckornfield) [inline .pre-commit-config.yaml:132]: If we add this here, do we need a separate lint job?

**Review by mckornfield — APPROVED** [review]
Just not sure if we're double linting since we run pre commit in ci

File: .pre-commit-config.yaml (hunk 3 above)
Comment (anastasia-nesterenko) [reply .pre-commit-config.yaml:132]: agreed, will remove and keep only pre-commit hook

Comment (anastasia-nesterenko) [reply .pre-commit-config.yaml]: Agreed. I removed the dedicated `skill-frontmatter-lint` CI job, its change-filter output, and its aggregate `CI status` dependency in 2c19f9a2c7. The checker and regression tests remain in the repository, and validation is now kept local through the pre-commit hook.

File: plugins/nemo-eval-author/skills/eval-author-audit/SKILL.md (hunk 2 above)
Comment (anastasia-nesterenko) [reply plugins/nemo-eval-author/skills/eval-author-audit/SKILL.md:34]: SkillEvaluator requires a scalar string, but its intended format is space-separated. The comma-separated constraint belongs to our local validator and, accidentally, SkillSpector—not SkillEvaluator’s schema. https://github.com/NVIDIA/SkillEvaluator/blob/main/src/skillevaluator/models/skill.py#L122-L126

**Review by crookedstorm — APPROVED** [review]
LGTM, better to get it before commit.
